applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClientBuilder.java

```diff
@@ -26,7 +26,29 @@
 import java.util.Objects;
 
 /**
- * Provides configuration options for instances of {@link ConfigurationAsyncClient}.
+ * This class provides a fluent builder API to help aid the configuration and instantiation of the {@link ConfigurationAsyncClient},
+ * calling {@link ConfigurationAsyncClientBuilder#build() build} constructs an instance of the client.
+ *
+ * <p>The client needs the service endpoint and access credentials to interact with Azure App Configuration. {@link ConfigurationClientCredentials}
+ * gives the builder the service endpoint and access credentials it required to construct a client, set the ConfigurationClientCredentials
+ * with {@link ConfigurationAsyncClientBuilder#credentials(ConfigurationClientCredentials) credentials}.</p>
+ *
+ * <pre>
+ * ConfigurationAsyncClient client = ConfigurationAsyncClient.builder()
+ *     .credentials(new ConfigurationClientCredentials(connectionString))
+ *     .build();
+ * </pre>
+ *
+ * <p>{@link HttpPipeline} gives the builder the access credentials it requires, set the HttpPipeline with {@link ConfigurationAsyncClientBuilder#pipeline(HttpPipeline) pipeline}.
+ * Using a HttpPipeline requires additional setup but allows for finer control on how the ConfigurationAsyncClient it built.
+ * Additionally, the service endpoint is also required, set the service endpoint with {@link ConfigurationAsyncClientBuilder#serviceEndpoint(String) serviceEndpoint}.</p>
+ *
+ * <pre>
+ * ConfigurationAsyncClient.builder()
+ *     .pipeline(new HttpPipeline(policies))
+ *     .serviceEndpoint(serviceEndpoint)
```

Comment:
Is this the consistently applied name (`serviceEndpoint`)? Check with Key Vault.

Reply:
KeyVault's equivalent is url, which I think it the better term to use as we reference URL when talking about it and the backing type is URL.

Reply:
I think "serviceEndpoint" makes it easier to interpret rather than just "url" (what URI are we referring to?) When you copy these values from portal.azure.com services like App Config, Storage, Cognitive Services call this value: "Endpoint" and "Service Endpoint"

Reply:
I would like to standardise here, yes. My initial instinct is to go with `endpoint`. I'm not sure that `serviceEndpoint` adds a lot more contextual information (that is, what other endpoint would we be referring to?). Can someone also please confer with the other languages to see where they have landed in this naming. Any other thoughts? // cc @g2vinay

Reply:
Appears .NET and Python use connectionString/connection_string and they extract the URL from that.

Reply:
Hm. That's one way to authenticate with Key Vault. You can also use a service principal, in which case there is no such thing as a connection string. You would use the keyvault URL, client ID and client secret.